### PR TITLE
[USM] Rate limit process monitor debug logs

### PR DESCRIPTION
**Pull request by Yumasi**
### What does this PR do?

Limit the number of debug logs the process monitor can generate, as they are currently spamming our logs in case the callback channels are full (we have seen number in the 9k logs per 5 minutes). Not only does it spam the logs, it causes massive allocations in the range of 150+MB when it happens.

This PR fixes this by introducing a log limiter so only 10 logs are allowed in a 10 minutes span.

### Motivation

Reduce log spam & memory consumption in case of callback channels being full, when debug logs are enabled.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by DanielLavie — APPROVED** [review]

File: pkg/process/monitor/process_monitor.go [hunk 1]
```diff
@@ -37,6 +38,8 @@ var (
 		processExecCallbacks: make(map[*ProcessCallback]struct{}, 0),
 		processExitCallbacks: make(map[*ProcessCallback]struct{}, 0),
 	}
+
+	oversizedLogLimit = util.NewLogLimit(10, 10*time.Minute)
```
Comment (guyarb) [inline pkg/process/monitor/process_monitor.go:42]: why is it global?

Comment (guyarb) [reply pkg/process/monitor/process_monitor.go:42]: it should be a member of the struct

Comment (guyarb) [reply pkg/process/monitor/process_monitor.go:42]: it should be a member of the struct

Comment (Yumasi) [reply pkg/process/monitor/process_monitor.go]: Fixed

**Pull request merged**